src/isomorphic/classic/class/ReactClass.js

```diff
@@ -437,6 +437,12 @@ function validateMethodOverride(isAlreadyDefined, name) {
  */
 function mixSpecIntoComponent(Constructor, spec) {
   if (!spec) {
+    warning(
+      typeof spec === 'undefined',
+      '%s: You\'re attempting to ' +
+      'include a mixin that is undefined.',
```

Comment:
If we're doing this, we might as well add checks for anything that is not an object.  So maybe we just wrap the whole thing in a `if(typeof spec !== 'object'){...}` and print the type in the error message.